packages/create-react-native-library/templates/native-common/android/build.gradle

```diff
@@ -69,28 +76,62 @@ android {
 <% if (project.arch === "new") { -%>
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
 <% } -%>
+<% if (project.cpp || project.nitro) { -%>
 
-<% if (project.cpp) { -%>
     externalNativeBuild {
       cmake {
-        cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all"
+        cppFlags "-frtti -fexceptions -Wall -fstack-protector-all"
+        arguments "-DANDROID_STL=c++_shared"
         abiFilters (*reactNativeArchitectures())
+
+        buildTypes {
+          debug {
+            cppFlags "-O1 -g"
+          }
+          release {
+            cppFlags "-O2"
+          }
+        }
```

Comment:
It doesnt _have_ to be configured like this. `-DANDROID_STL=c++_shared` is something I recommend for EVERY C++ lib, even Turbo/Expo modules, but the rest is purely up to you guys. Leaving it like this is good imo, but it's technically slower in debug because of -O1

Reply:
@mrousavy it's going to run slower, not build slower right? Besides that I think this is a great default to have for every cpp lib, that's why I kept it this way.

Reply:
It's gonna build faster, and run slower in debug (`-O1` is less aggressive optimizations than `-O2`). :)

This also might include some more debug symbols and strips out less unused vars, might help for debugging.

Reply:
I think that would be perfectly reasonable in this case, also builds running faster on debug would be the cherry on the top